crates/matrix-sdk/src/account.rs

```diff
@@ -1188,10 +1192,17 @@ impl Account {
 
         let index = recent_emojis.iter().position(|(unicode, _)| unicode == emoji);
 
+        // Remove the emoji from the list if it was present and get it's `count` value
         let count = if let Some(index) = index { recent_emojis.remove(index).1 } else { uint!(0) };
 
+        // Insert the emoji with the updated count at the start of the list, so it's
+        // considered the most recently used emoji
```

Comment:
The `insert` is `O(n)`, so if we're worried about the size, perhaps we should truncate before the insert.